app/controllers/api/v1/search_controller.rb

```diff
@@ -207,6 +208,164 @@ def facet_filters
         @matching_filters = @search_facet.filters.select {|filter| filter[:name] =~ query_matcher}
       end
 
+      swagger_path '/search/auth_code' do
+        operation :post do
+          key :tags, [
+              'Search'
+          ]
+          key :summary, 'Create One-time Auth Code for downloads'
+          key :description, 'Create and return a One-Time Authorization Code to identify a user for bulk downloads'
+          key :operationId, 'search_auth_code_path'
+          response 200 do
+            key :description, 'One-time auth code and time interval, in seconds'
+            schema do
+              property :totat do
+                key :type, :integer
+                key :description, 'One-time auth code'
+              end
+              property :ti do
+                key :type, :integer
+                key :description, 'Time interval (in seconds) otac will be valid'
+              end
+            end
+          end
+          response 401 do
+            key :description, 'User is not signed in'
+          end
+          response 406 do
+            key :description, 'Accept or Content-Type headers missing or misconfigured'
+          end
+        end
+      end
+
+      def create_auth_code
+        half_hour = 1800 # seconds
+        otac_and_ti = current_api_user.create_totat(half_hour)
+        render json: otac_and_ti
+      end
+
+      swagger_path '/search/bulk_download' do
+        operation :get do
+          key :tags, [
+              'Search'
+          ]
+          key :summary, 'Bulk download study data'
+          key :description, 'Download files in bulk of multiple types from one or more studies via curl'
+          key :operationId, 'search_bulk_download_path'
+          parameter do
+            key :name, :auth_code
+            key :type, :integer
+            key :in, :query
+            key :description, 'User-specific one-time authorization code'
+            key :required, true
+          end
+          parameter do
+            key :name, :accessions
+            key :type, :string
+            key :in, :query
+            key :description, 'Comma-delimited list of Study accessions'
+            key :required, true
+          end
+          parameter do
+            key :name, :file_types
+            key :type, :string
+            key :in, :query
+            key :description, "Comma-delimited list of file types (including 'all' for all files)"
```

Comment:
Yeah - Swagger can do enums.  I'll add that in.  And I like that approach for the parameter- if there are no specified file types, all will be returned.

Reply:
So I'm hesitant to do the snake_case representation of the file types in the parameter, because it will require converting back and forth between this format and the format that is stored in the `file_type` attribute for each `StudyFile` in order to do the query.  If the only reason for this is to make the query parameters easier to pass, I would argue it's just as easy to pass the normal form than snake case, and it makes the Mongo query easier.